virtcontainers/qemu.go

```diff
@@ -1230,7 +1249,33 @@ func (q *qemu) addDevice(devInfo interface{}, devType deviceType) error {
 
 	switch v := devInfo.(type) {
 	case types.Volume:
-		q.qemuConfig.Devices = q.arch.append9PVolume(q.qemuConfig.Devices, v)
+		if q.config.VirtioFS {
+			q.Logger().WithField("volume-type", "virtio-fs").Info("adding volume");
+
+			randBytes, err := utils.GenerateRandomBytes(8)
+			if err != nil {
+				return err
+			}
+			id := hex.EncodeToString(randBytes)
+
+			// TODO make this unique so multiple volumes can be added!
+			sockPath, err := utils.BuildSocketPath(runStoragePath, q.id, "vhost-fs.sock")
+			if err != nil {
+				return err
+			}
+
+			vhostDev := config.VhostUserDeviceAttrs{
+				Tag: v.MountTag,
+				Type: config.VhostUserFS,
```

Comment:
Indent seems to be off here. @stefanha Can you format the files using `gofmt`?

Reply:
Thanks, will fix in the next revision.